### PR TITLE
Adds missing plugins to SPIRE docs index

### PR DESCRIPTION
**Pull request by ajessup**
**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [X] Documentation updated?

**Description of change**
What says on the tin. This is a doc fix only.

### CODE REVIEW

File: doc/spire_agent.md [hunk 1]
```diff
@@ -83,9 +83,12 @@ communicates with spire-server via the Node API.
...
 | NodeAttestor     | [join_token](/doc/plugin_agent_nodeattestor_jointoken.md) | A node attestor which uses a server-generated join token |
-| NodeAttestor     | [aws_iid](/doc/plugin_agent_nodeattestor_aws_iid.md) | An AWS IID attestor that automatically attests instances using the AWS Instance Metadata API and the AWS Instance Identity document. |
+| NodeAttestor     | [x509_pop](/doc/plugin_agent_nodeattestor_x509pop.md) | A node attestor which uses a pre-existing x.509 certificate and key |
+| NodeAttestor     | [aws_iid](/doc/plugin_agent_nodeattestor_aws_iid.md) | An AWS IID attestor that automatically attests EC2 instances using the AWS Instance Metadata API and the AWS Instance Identity document. |
+| NodeAttestor     | [azure_msi](/doc/plugin_agent_nodeattestor_azure_msi.md) | An Azure Node attestor that automatically attests Azure VMs using a signed Managed Service Identity (MSI) token. |
+| NodeAttestor     | [gcp_iid](/doc/plugin_agent_nodeattestor_gcp_iid.md) | An GCP IID attestor that automatically attests GCE instances using a signed token from Google retrieved via the Compute Engine Metadata API. |
```
Comment (amartinezfayo) [inline doc/spire_agent.md]: The name is `gcp_iit`. This affects the link also.

File: doc/spire_server.md [hunk 2]
```diff
@@ -167,7 +167,11 @@ API and the Node API, with which agents communicate with the server.
...
+| NodeResolver | [noderesolver_aws_iid](/doc/plugin_server_noderesolver_aws_iid.md) | A node resolver which extends the [aws_iid](/doc/plugin_server_nodeattestor_aws_iid.md) node attestor plugin to support selecting nodes based on additional properties (such as Security Group ID). |
+| NodeAttestor | [nodeattestor_azure_msi](/doc/plugin_server_nodeattestor_azure_msi.md) | A node attestor which validates agents attesting using the [azure_msi](/doc/plugin_agent_nodeattestor_azure_msi.md) node attestor plugin. |
+| NodeResolver | [noderesolver_azure_msi](/doc/plugin_server_noderesolver_aws_iid.md) | A node resolver which extends the [nodeattestor_azure_msi](/doc/plugin_server_nodeattestor_azure_msi.md) node attestor plugin to support selecting nodes based on additional properties (such as Network Security Group). |
+| NodeAttestor | [nodeattestor_gcp_iid](/doc/plugin_server_nodeattestor_gcp_iid.md) | A node attestor which validates agents attesting using the [gcp_iid](/doc/plugin_agent_nodeattestor_gcp_iid.md) node attestor plugin. |
```
Comment (amartinezfayo) [inline doc/spire_server.md]: The name is `gcp_iit`. This affects the links also.

**Comment by amartinezfayo** [comment]
Looks like the sign-off is missing in the commits.

**Comment by ajessup** [comment]
Fixed DCO. @amartinezfayo can you re-review?

**Pull request merged**